pkg/network/prober/prober.go

```diff
@@ -30,16 +30,43 @@ import (
 	"github.com/knative/serving/pkg/network"
 )
 
-// TransportFactory is a function which returns a HTTP transport
-type TransportFactory func() http.RoundTripper
+type Prober interface {
+	Do(ctx context.Context, target, headerValue string, pos ...ProbeOption) (bool, error)
+	Offer(ctx context.Context, target, headerValue string, callback Callback, period, timeout time.Duration) bool
+}
+
+// Ensure Manager implements Prober interface
+var _ Prober = &Manager{}
 
 // ProbeOption is a way for caller to modify the HTTP request before it goes out.
 type ProbeOption func(r *http.Request) *http.Request
 
+// Callback is executed when the an async probe completes.
+type Callback func(success bool, err error)
+
+// Manager manages async probes and makes sure we run concurrently only a single
+// probe for the same key.
+type Manager struct {
+	RoundTripper http.RoundTripper
```

Comment:
No this doesn't work.
There is a reason why prober needs factory, rather than just a roundtripper. Please retain that.

Reply:
Ok. What did I miss?